community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/CursorRelationshipIterator.java

```diff
@@ -71,12 +72,14 @@ public long next()
         {
             try
             {
-                id = cursor.getId();
-                type = cursor.getType();
-                startNode = cursor.getStartNode();
-                endNode = cursor.getEndNode();
+                // Copy is necessary here as the nextCursor() in finally will change contents of it
+                RelationshipItem item = cursor.get();
+                id = item.getId();
+                type = item.getType();
```

Comment:
id, type, startNode and endNode fields probably should be cleaned after `hasNext()` returned false or `NoSuchElementException` was raised